### ISSUE TITLE
less restrictive dependency pinning

### ISSUE THREAD

**Issue by jannikmi** [huggingface/setfit#232]
Currently the setup.py is pinning specific dependency versions (cf. https://github.com/huggingface/setfit/blob/dfa4541af1a4166e76f5b21595b51c1a56324d6b/setup.py#L11).
This is not best practice as it makes dependency conflicts super likely.

Please change the dependency specification to something more permissive like:

```
REQUIRED_PKGS = ["datasets>=2.3.2", "sentence-transformers>=2.2.2", "evaluate>=0.3.0"]
```

**Comment by tomaarsen** [comment]
I'm in favour. I think `datasets>=2.3.2` is likely not even needed - it's probably just what was being used when the package was originally developed: `datasets>=2.0.0` probably works too. 

I'd be interested in a CI workflow that runs tests on the oldest allowed version in order to let us allow the widest range of versions.

**Comment by jannikmi** [comment]
[poetry](https://python-poetry.org/docs/dependency-specification/) offers this kind of dependency management by pinning specific package versions in a separate .lock file (useful for testing as @tomaarsen mentioned) while keeping the general "public" dependency versions permissive

see for example: https://github.com/jannikmi/timezonefinder/blob/master/pyproject.toml

**Comment by tomaarsen** [comment]
Although I'm fairly happy with poetry for the few projects that I've used it in, I don't see myself changing the existing setup for this. The less restrictive dependencies can be done using the simple `setup.py` too, and I don't want to stray too far from the other HF products in terms of project structure.  

That said, I do think updating the dependencies is important. As a result, I opened a PR that increases the leniency of the dependencies.

- Tom Aarsen
